zict/sieve.py

```diff
@@ -1,10 +1,16 @@
+from __future__ import annotations
+
 from collections import defaultdict
+from collections.abc import Callable, Iterable, Iterator, Mapping, MutableMapping
```

Comment:
in common.py and zip.py, `MutableMapping` is imported from the `typing` module for compatibility with python <3.9. 

But I think `MutableMapping` has been in python for a while now:
https://docs.python.org/3.7/library/collections.abc.html#collections.abc.MutableMapping

Do we rely on a specific implementation that's only available in > Python3.8?

Reply:
In Python 3.9+, collections.abc.MutableMapping introduces support for square brackets like typing.MutableMapping does.
In other places we're importing from collections.abc; that's because those are modules where MutableMapping[...] is used exclusively in delayed type annotations, which are not parsed by the python interpreter.